delivery_postlogistics/postlogistics/web_service.py

```diff
@@ -202,21 +195,18 @@ def _prepare_attributes(
             "przl": services,
             "weight": total_weight,
         }
-        option_codes = [
-            packaging.postlogistics_option_code
-            for packaging in postlogistics_packagings
-        ]
+        option_code = packaging.postlogistics_option_code
 
-        if "ZAW3217" in option_codes and picking.delivery_fixed_date:
+        if option_code == "ZAW3217" and picking.delivery_fixed_date:
             attributes["deliveryDate"] = picking.delivery_fixed_date
-        if "ZAW3218" in option_codes and pack_num:
+        if option_code == "ZAW3218" and pack_num:
```

Comment:
ZAW3218 is not related to parcel number.

Replace with:
`if pack_num:`

Reply:
fixed in https://github.com/OCA/delivery-carrier/pull/323/commits/f04b053b0aeca3c51756f5ad9129349ce26f4555